include/Basic/SourceManager.hpp

```diff
@@ -0,0 +1,195 @@
+#ifndef GLU_SOURCE_MANAGER_HPP
+#define GLU_SOURCE_MANAGER_HPP
+
+#include <llvm/ADT/SmallVector.h>
+#include <llvm/Support/MemoryBuffer.h>
+#include <optional>
+
+#include "FileEntry.hpp"
+#include "SourceLocation.hpp"
+#include "llvm/ADT/StringRef.h"
+
+namespace glu::srcmanager {
```

Comment:
namespace glu {} for the SourceManager
you can use an anonymous namespace for the rest, if it is only used within the file